x-pack/plugins/cases/server/client/configure/validators.ts

```diff
@@ -38,3 +48,74 @@ export const validateCustomFieldTypesInRequest = ({
     );
   }
 };
+
+/**
+ * Throws if the template key doesn't match the configuration
+ */
+export const validateTemplateKeysAgainstConfiguration = ({
+  requestTemplateFields,
+  templatesConfiguration,
+}: {
+  requestTemplateFields: TemplatesConfiguration | undefined;
+  templatesConfiguration: TemplatesConfiguration;
+}) => {
+  if (!Array.isArray(requestTemplateFields) || !requestTemplateFields.length) {
+    return;
+  }
+
+  if (templatesConfiguration === undefined) {
+    throw Boom.badRequest('No templates configured.');
+  }
+
+  if (!templatesConfiguration.length) {
+    return;
+  }
+
+  const invalidTemplateKeys = differenceWith(
+    requestTemplateFields,
+    templatesConfiguration,
+    (requestVal, configurationVal) => requestVal.key === configurationVal.key
+  ).map((e) => e.key);
+
+  if (invalidTemplateKeys.length) {
+    throw Boom.badRequest(`Invalid template keys: ${invalidTemplateKeys}`);
+  }
+};
+
+export const validateTemplatesCustomFieldsInRequest = ({
+  templates,
+  customFieldsConfiguration,
+}: {
+  templates?: TemplatesConfiguration;
+  customFieldsConfiguration?: CustomFieldsConfiguration;
+}) => {
+  if (!Array.isArray(templates) || !templates.length) {
+    return;
+  }
+
+  templates.forEach((template, index) => {
+    if (
+      !template.caseFields ||
+      !template.caseFields.customFields ||
+      !template.caseFields.customFields.length
+    ) {
+      return;
+    }
+
+    if (customFieldsConfiguration === undefined) {
+      throw Boom.badRequest('No custom fields configured.');
+    }
+
+    const params = {
+      requestCustomFields: template.caseFields.customFields,
+      customFieldsConfiguration,
```

Comment:
This will not work when we have multiple templates with same customFields to update.
i.e. if In patch request, `template_1` has `custom_field_1` and `template_2` also has `custom_field_1` then it will throw duplicate key error form custom field keys because we have gathered all customFields in single array.

Reply:
oh, good point! We need to verify per template.